### PR TITLE
Disable default assignment for Unique Columns

### PR DESCRIPTION
**Pull request by Aritra8438**
**This PR is related to** #2821 
Primarily, this PR disables ```default``` assignment when the **column has uniqueness constraint**.

## Technical details
I have handled the issue in the following way:
- When a column is restricted to unique, default block won't be rendered. **(after toggling, default block will be available again.)**
- If a column has a default value, **restricting the same to unique will unset the default** (otherwise new record insertion will eventually violate the uniqueness constraint).
- Default value modification of unique columns will not be allowed via **the API through ```PATCH``` request**.

So, **this PR has both UI and backend changes.**
## Screencast:

https://user-images.githubusercontent.com/64671908/234174295-bb87cf57-8838-43ce-bd36-34c0d2017193.mp4

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by Aritra8438** [comment]
Hi, @silentninja, @rajatvijay, I'm making this PR draft for a while. 

There was a similar PR (#2780) and some of the issues mentioned by @silentninja need to be addressed here as well. I will mark it ready for review after I address those changes.

**Comment by Aritra8438** [comment]
Hi, @rajatvijay. Sorry for being late.

Two prerequisite PRs have been merged and I will start working on this.
Please keep this issue assigned to me.

File: mathesar_ui/src/systems/table-view/table-inspector/column/ColumnOptions.svelte [hunk 1]
```diff
@@ -58,26 +58,30 @@
...
+      .catch((error) => {
+        const message = `Unable to update "Allow Duplicates" of column "${
+          column.column.name
+          // @ts-ignore: https://github.com/centerofci/mathesar/issues/1055
+        }". ${error.message as string}.`;
+        toast.error({ message });
+      })
+      .finally(() => {
+        isRequestingToggleAllowDuplicates = false;
+      });
```
Comment (rajatvijay) [inline mathesar_ui/src/systems/table-view/table-inspector/column/ColumnOptions.svelte:84]: As per the FE standards here: https://github.com/centerofci/mathesar/blob/develop/mathesar_ui/STANDARDS.md#await-vs-then, async/await is preferred here. Was there any specific reason to change?

**Review by rajatvijay — CHANGES_REQUESTED** [review]
@Aritra8438 Made a comment from the FE side. 

@silentninja ninja will be taking care of the review from BE side.

**Comment by mathemancer** [comment]
I can come up with examples where you'd want a static default in a column with a unique constraint. I think we should not implement this functionality.
- You may have time series data where you want at most one 'current' row, given as the default 1970-01-01. I've seen this in real life.
- You may want to use the default in some kind of odd 'on conflict do update' control scheme.

Big picture, this is allowed by Postgres, and I think we should allow it as well. We should, however, double-check that we handle any resulting error on insert gracefully.

**Pull request closed without merging**